app/controllers/vocabulary_controller.php

```diff
@@ -63,14 +63,27 @@ public function of($username, $lang = null)
     {
         $this->helpers[] = 'Pagination';
         $this->helpers[] = 'CommonModules';
+
+        $username = Sanitize::paranoid($username, array('_'));
         
         $userId = $this->User->getIdFromUsername($username);
+
+        if (!$userId) {
+            $this->Session->setFlash(
+                __('No user with this username: ', true).$username
```

Comment:
`$username` should be included in the translated string as a placeholder, so that translators are not forced to put it at the end of the sentence. Besides, it’s also currently unclear for the translators that the string is to be followed by a username. You can find some code examples [here](https://github.com/Tatoeba/tatoeba2/search?utf8=%E2%9C%93&q=no+such+user).